pkg/virt-launcher/virtwrap/agent-poller/agent_parser.go

```diff
@@ -254,31 +254,53 @@ func convertInterfaceStatusesFromAgentJSON(agentResult []Interface) []api.Interf
 			continue
 		}
 
-		interfaceIP, interfaceIPs := extractIPs(ifc.IPs)
+		var ip, cidr string
+		ips, cidrs := extractNetwork(ifc.IPs)
+		if len(ips) > 0 {
+			ip = ips[0]
+		}
+		if len(cidrs) > 0 {
+			cidr = cidrs[0]
+		}
+
 		interfaceStatuses = append(interfaceStatuses, api.InterfaceStatus{
 			Mac:           ifc.MAC,
-			Ip:            interfaceIP,
-			IPs:           interfaceIPs,
+			Ip:            ip,
+			IPs:           ips,
+			CIDR:          cidr,
+			CIDRs:         cidrs,
 			InterfaceName: ifc.Name,
 		})
 	}
 	return interfaceStatuses
 }
 
-func extractIPs(ipAddresses []IP) (string, []string) {
+func extractNetwork(ipAddresses []IP) ([]string, []string) {
 	interfaceIPs := []string{}
+	interfaceCIDRs := []string{}
+
 	var interfaceIP string
+	var interfaceCIDR string
+
 	for _, ipAddr := range ipAddresses {
-		ip := fmt.Sprintf("%s/%d", ipAddr.IP, ipAddr.Prefix)
+		ip := ipAddr.IP
+		cidr := fmt.Sprintf("%s/%d", ip, ipAddr.Prefix)
 		// Prefer ipv4 as the main interface IP
 		if ipAddr.Type == "ipv4" && interfaceIP == "" {
 			interfaceIP = ip
+			interfaceCIDR = cidr
+		} else {
+			interfaceIPs = append(interfaceIPs, ip)
+			interfaceCIDRs = append(interfaceCIDRs, cidr)
 		}
-		interfaceIPs = append(interfaceIPs, ip)
 	}
-	// If no ipv4 interface was found, set any IP as the main IP of interface
-	if interfaceIP == "" && len(interfaceIPs) > 0 {
-		interfaceIP = interfaceIPs[0]
+
+	// If ipv4 interface was found, set the IP as the first element.
+	// The first element will be set as a primary address.
```

Comment:
Just a nit. I don't like the fact you specify here how the method should be used. How can you assure the first element will be used as the primary address? I prefer the method will return - ip, ips, cidr, cidrs.

Reply:
as you wish, I will change that

Reply:
> I don't like the fact you specify here how the method should be used. How can you assure the first element will be used as the primary address?

And.. just for clarity and explanation. I think I don't force anybody to use this method in a different way that was done before. The overall result is still the same. It's just a matter of returning primary address as a first element of list (which is assured by prepend) or returning it as a new value, but I think I get your point as returning a new value is explicit.